### PR TITLE
Feat/home hero responsive images #6411

### PR DESCRIPTION
**Pull request by creido-welly**
- added switching of low-res to high-res image sources after Hero component has been initially rendered on client
- Data structure updated for flexibility i.e. an array of data will allow us to populate a variable number of `<source>` elements within a `<picture>` element
- responsive image markup extracted as separate Picture component

### CODE REVIEW

File: src/components/SlideshowHero/_slideshow-hero.scss [hunk 1]
```diff
@@ -194,10 +194,22 @@
...
+    height: 100%;
+    object-fit: cover;
+  }
+
+  @media all and (-ms-high-contrast:none) {
+    top: 50%;
+    transform: translateY(-50%);
+  }
+
+  &.lazy {
```
Comment (chriddell) [inline src/components/SlideshowHero/_slideshow-hero.scss]: Wondering about this classname... thinking should it be:

- `is-lazy` to denote a state... but it's not stateful really... so should it be
- `.slideshow-image--lazy` to denote modifier?

**Review by chriddell — CHANGES_REQUESTED** [review]

File: src/components/SlideshowHero/_slideshow-hero.scss (hunk 1 above)
Comment (creido-welly) [reply src/components/SlideshowHero/_slideshow-hero.scss]: `.u-lazy` prob better since (I hope) it'll become universal

thoughts?

Comment (lukewhitmore) [reply src/components/SlideshowHero/_slideshow-hero.scss]: Will a hero slideshow ever need lazy loading if it's always above the fold?

Comment (chriddell) [reply src/components/SlideshowHero/_slideshow-hero.scss]: Guess it's more "progressive" than lazy, in that we're loading a low-res placeholder to begin with to make the UI feel fast.

> .u-lazy 

Don't think it's a utility classname as it is, unless we want to change the name to `u-blur` to be more descriptive?

Comment (creido-welly) [reply src/components/SlideshowHero/_slideshow-hero.scss]: Agreed, it is above the fold so ideally shouldn't be called lazy loaded and it is indeed more progressive load.

Blur is more descriptive of the styling effect which gets applied whereas "preloading" is a better descriptor of what's happening technically. In which case how about one of the following?:

`.has-preload`
`.is-preload`
`.is-preloaded`

Comment (chriddell) [reply src/components/SlideshowHero/_slideshow-hero.scss]: the `has/is` would work for now, but it restricts it to only the `slideshow__image` selector (`.slideshow__image.is-preload`)

so if we wanted it to be universal selector it should be a utility class imo, and become something (e.g.) `u-img-preload`

Comment (creido-welly) [reply src/components/SlideshowHero/_slideshow-hero.scss]: sold!

File: src/components/SlideshowHero/_slideshow-hero.scss [hunk 2]
```diff
@@ -194,10 +194,22 @@
...
+  }
+
+  @media all and (-ms-high-contrast:none) {
+    top: 50%;
+    transform: translateY(-50%);
+  }
+
+  &.u-loading-preload {
+    filter: blur(20px) brightness(110%);
+  }
```
Comment (chriddell) [inline src/components/SlideshowHero/_slideshow-hero.scss]: not sure about the nesting... how about adding the `u-loading-preload` selector to https://github.com/wellcometrust/corporate-components/tree/develop/src/assets/styles/50-utilities?

I think if the styles are specific to the `slideshow__image`, then use `&.is-loading-preload`, but if they're more agnostic then add to util partials

Comment (creido-welly) [reply src/components/SlideshowHero/_slideshow-hero.scss]: Done

Comment (chriddell) [reply src/components/SlideshowHero/_slideshow-hero.scss]: ace

**Review by chriddell — APPROVED** [review]

**Pull request merged**